src/factorization.jl

```diff
@@ -144,9 +144,42 @@ function restarted_arnoldi(A::AbstractMatrix{T}, min = 5, max = 30, tolerance =
     for restarts = 1 : max_restarts
         iterate_arnoldi!(A, arnoldi, min + 1 : max)
         implicit_restart!(arnoldi, min, max, active)
-        active = detect_convergence!(view(arnoldi.H, 1:min+1, 1:min), tolerance)
+
+        new_active = detect_convergence!(view(arnoldi.H, 1:min+1, 1:min), tolerance)
+
+        # Bring the new locked part oF H into upper triangular form
+        if new_active > active 
```

Comment:
This looks good to me :)

Reply:
Quick note, what if only 1 eigenvector converges (`new_active == active + 1`)? In that case we probably shouldn't do anything, cause we end up with a 2x1 block with the bottom value 0 (or well, it's upper triangular already.) Skip over that case then?